### PR TITLE
xen: update to 4.14.2.

### PR DESCRIPTION
**Pull request by CMB**
- xen: update to 4.14.2.

#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [ x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

**Comment by ericonr** [comment]
@void-linux/pkg-committers could someone help with review here?

**Comment by ericonr** [comment]
@void-linux/pkg-committers could someone help with review here?

File: srcpkgs/xen/files/xenconsoled/check
Comment (heliocat) [inline srcpkgs/xen/files/xenconsoled/check:15]: Consider changing this to `lsof -p ${PID:-0} | grep -q hypervisor.log` or rolling this and the PID= line into a single command. It still isn't perfect but should be less fiddly.

File: srcpkgs/xen/files/xen/run
Comment (heliocat) [inline srcpkgs/xen/files/xen/run:2]: `sv start` wraps `sv check` so you can probably remove the check line.

File: srcpkgs/xen/files/xenconsoled/run
Comment (heliocat) [inline srcpkgs/xen/files/xenconsoled/run:11]: I would move this to `xenconsoled/log/run` unless xenconsoled/run is the process that writes to it.

File: srcpkgs/xen/files/xenconsoled/run
Comment (heliocat) [inline srcpkgs/xen/files/xenconsoled/run:4]: Same commentary wrt removing check as `xen/run`

File: srcpkgs/xen/files/xenconsoled/run
Comment (heliocat) [inline srcpkgs/xen/files/xenconsoled/run:6]: Same commentary as `xen/run` wrt pgrep vs looking in `supervise/pid.

File: srcpkgs/xen/files/xenstored/check
Comment (heliocat) [inline srcpkgs/xen/files/xenstored/check:7]: Ditto

File: srcpkgs/xen/files/xenstored/run
Comment (heliocat) [inline srcpkgs/xen/files/xenstored/run:13]: Style preference: don't leak shellcheck comments into run scripts (arguably, shellcheck complaining about a guarded file is a bug).

File: srcpkgs/xen/patches/argp-standalone.patch
Comment (heliocat) [inline srcpkgs/xen/patches/argp-standalone.patch:4]: Change directory name to xen-4.14.2?

File: srcpkgs/xen/files/xenstored/run
Comment (ahesford) [reply srcpkgs/xen/files/xenstored/run:13]: I second, don't pollute the run scripts with spellcheck noise.

File: srcpkgs/xen/files/xenstored/run
Comment (ahesford) [inline srcpkgs/xen/files/xenstored/run:25]: Again, no shell check noise

File: srcpkgs/xen/files/xenstored/run
Comment (ahesford) [inline srcpkgs/xen/files/xenstored/run:26]: Add `--no-fork` here rather than to the variable, because it will always be required and we don't want to make people who customize the args have to remember to include it.

File: srcpkgs/xen/files/xenstored/run
Comment (ahesford) [inline srcpkgs/xen/files/xenstored/run:16]: Avoid the sub shell, `${XENSTORED##*/}` works in place of base name.

Better yet, why does this case exist? Just drop the whole test, you already provide a mechanism for people to add `--verbose` In `$XENSTORED_ARGS` if they care.

File: srcpkgs/xen/files/xenstored/run
Comment (ahesford) [inline srcpkgs/xen/files/xenstored/run:9]: 1. Don't make no-op services. If this isn't dom0, people shouldn't enable the service. Just fail here.
2. Don't pollute the process name with documentation like this.

**Review by ahesford — CHANGES_REQUESTED** [review]
I didn't exhaustively comment on the service files, but left a few points. In addition, consider these broad points:

1. Never do no-op services. If the service only works in dom0, the user must be responsible enough to only enable it in dom0. Just fail in domU.
2. If your no-op services aren't really no-op but actually do some upfront, one-shot work even in domU, either leave this to the user to put in`rc.local` or, if there is a really compelling reason, split into a "one-shot" that can be used in both domains. (Often there is no truly compelling reason for these one-shots.)
3. Like heliocat says, don't do pgrep or other process hacks. Unlike heliocat, I think using `/proc` is just as ugly. Don't do it.
4. Don't `sv start` from another service. Runit isn't good about dependencies, and these kind of hacky workarounds make the situation worse, not better. You can `sv check` if you really need to, but *only* if the service you check has a proper `check` script to meaningfully determine whether the service is sufficiently up. In that case, just fail if the dependency fails its check.

Rework the scripts in light of these and more specific comments and let's see where we stand.

File: srcpkgs/xen/files/xenconsoled/check
Comment (ahesford) [reply srcpkgs/xen/files/xenconsoled/check:15]: If this check really matters, and there's no reliable way to actually determine whether the service is running, might as well just use `chpst -L` in the run script to acquire a lock on some file like `/run/xen/xenconsoled.lock`. Then the check script can just try to acquire that lock and, if it succeeds, you know the service is not running.

Yes, this would be a hack. It would also be much less hacky than trying to scrape the process tree.

**Review by ahesford — CHANGES_REQUESTED** [review]

File: srcpkgs/xen/files/xenconsoled/check
Comment (heliocat) [reply srcpkgs/xen/files/xenconsoled/check:15]: Yeah, a lock works too. I don't particularly like them in shell scripts but `chpst -L lockfile ...` in the run script and `flock -n lockfile` in the check script works as well.

File: srcpkgs/xen/files/xenstored/run
Comment (heliocat) [reply srcpkgs/xen/files/xenstored/run:16]: I had to look it up, it's picking between two different xenstored binaries, both of which are shipped with the package and I believe take different slightly different options.

**Comment by heliocat** [comment]
>     3. Like heliocat says, don't do pgrep or other process hacks. Unlike heliocat, I think using `/proc` is just as ugly. Don't do it.
> 
I didn't say I thought `/proc` wasn't ugly, more that it's less ugly to use the native pid following in runit and then check the specific process details than it is to walk the entire process tree. The `chpst -L` route is probably the least-bad since it'll be relatively TOCTOU-free.

File: srcpkgs/xen/files/xenstored/run
Comment (ahesford) [reply srcpkgs/xen/files/xenstored/run:16]: Yes, but the only difference in the two cases is a default `--verbose` in one. Unless that is a required argument for `oxenstored` (which would shock me), it seems the case as used is superfluous.

**Comment by ahesford** [comment]
> I didn't say I thought `/proc` wasn't ugly, more that it's less ugly to use the native pid following in runit and then check the specific process details than it is to walk the entire process tree.

"[J]ust as ugly" is too strong; I agree using `/proc` isn't as bad as `pgrep`.

All of these are racy because no solutions guarantee that the daemon is in a usable state when the check is applied. I suppose a key question is what happens with `xenstore-write` if the necessary services aren't up. If the process just fails detectably, then drop the `check` script and the `sv check` and just monitor the return code on the `xenstore-write` invocations. Otherwise, if `xenstore-write` does unexpected things, the lock should accomplish what we need.

File: srcpkgs/xen/files/xenstored/run
Comment (heliocat) [reply srcpkgs/xen/files/xenstored/run:16]: Good point. I made it as far as reading the documentation that noted that there were different binaries but didn't manage to find the option reference. I can't say I looked particularly hard so it could be that it's as minor a difference as that.

File: srcpkgs/xen/files/xenconsoled/check
Comment (Piraty) [reply srcpkgs/xen/files/xenconsoled/check:13]: there are a few run scripts that exit early if dbus is not up which you could use for inspiration

**Comment by CMB** [comment]
@heliocat @ahesford Thanks for reviewing this, and I'm sorry for taking
so long to circle back to it.  I think all the issues you pointed out are now fixed.

**Comment by CMB** [comment]
Also orphaning.  I do like the Xen tooling, but I want to get rid of another
moving part in my environment.

**Pull request closed without merging**
